src/shaders/pbrMaterials.glsl

```diff
@@ -145,7 +145,9 @@ PBRData buildPBRData() {
   pbrInfo.view = normalize(uCameraWorldPos - position);
 
   // cos angle between view and normal
-  pbrInfo.n_dot_v = abs(dot(pbrInfo.n, pbrInfo.view));
+  // clamp to ensure range adherence, abs instead of epsilon to avoid errors at
+  // clancing angles
+  pbrInfo.n_dot_v = clamp(dot(pbrInfo.n, pbrInfo.view), 0.0, 1.0);
```

Comment:
I was experimenting with using the absolute value of the normal dot view instead of the clamp, and forgot to revert it. This in general looked just as good as the clamp, except in certain situations where it seems negative normal dot view results were indicated from a texture to give the object some kind of depth. In this case, the abs(n_dot_v) ended up looking weird.

First, an example of negative n_dot_v - frags with negative n_dot_v are the brightly colored teal->yellow, with the more yellow having the higher abs magnitude


![ndotvNegative](https://github.com/facebookresearch/habitat-sim/assets/1093953/4370112f-74ba-439a-a456-d98ddddf8162)

Here's what this looks like if we just leave the negative n_dot_v  : 

![ndotvNegNoGreen](https://github.com/facebookresearch/habitat-sim/assets/1093953/2cdd9976-1216-4302-bbf3-22365cb3c053)

Here's what the abs(n_dot_v) yielded - more character/variation in the area with the negative n_dot_v, but doesn't look quite right, spatially.

![absNdotV](https://github.com/facebookresearch/habitat-sim/assets/1093953/50169ada-e4a2-447b-b812-117fa218fc42)

And here's the clamped negative n_dot_v to [0,1]  - the inner area is a bit ambiguous, but it at least looks appropriate, geometrically.

![clampedNdotV](https://github.com/facebookresearch/habitat-sim/assets/1093953/717ea8cd-42ab-46a1-9a92-1276efca3673)

I don't clamp the normal to epsilon because that introduced artifacts around the edges. There are guards against zero n_dot_v where they would be needed.

Reply:
I thought it would be appropriate to chronicle one of the fundamental design decisions here (how to precondition n_dot_v) .